cargo-miri/bin.rs

```diff
@@ -456,6 +456,10 @@ fn phase_setup_rustc(args: env::Args) {
     let mut cmd = miri();
     cmd.args(args);
 
+    if !is_target_crate() {
+        cmd.env("MIRI_BE_RUSTC", "host");
+    }
```

Comment:
Why this? The comment above explicitly says "MIRI_BE_RUST is already set".

Reply:
This sets it to `"host"` instead of `"target"`. I think this can removed if the `OutputMirHack` is dropped, anyway.

Reply:
Are you saying "target" is wrong? That would be caught by our existing tests though, wouldn't it?

Reply:
Oh, the actual problem is in `setup` which builds the sysroot and indiscriminantly sets `MIRI_BE_RUSTC` to "target". That's just wrong, but should probably be fixed in a separate PR.

Reply:
It could be wrong, but it actually becomes problematic after I add this (IIRC):
```rust
            // Avoid that "crate required to be available in rlib format" error.
            config.opts.crate_types = vec![CrateType::Rlib];
```

Reply:
Yeah previously I guess the stdlib build scripts don't care that we are compiling them in "target" mode.